File: Algorithms/Sorting/BubbleSort.java

import java.util.Scanner;

class Main
{
    public static void main(String []args) {
        int n, temp;
        Scanner in = new Scanner(System.in);

        System.out.print("Enter Total number of elements: ");
        n = in.nextInt();

        int array[] = new int[n];
        System.out.println("Enter " + n + " integers:\n");
        for(int i=0; i<n; i++)
            array[i] = in.nextInt();

        for(int i=0; i<n-1; i++) {
            for(j=0; j<n-i-1; j++) {
                /* For descending order use < */
                if (array[j] > array[j+1]) {
                    temp       = array[j];
                    array[j]   = array[j+1];
                    array[j+1] = temp;
                }
            }
        }

        System.out.println("Sorted list of numbers:\n");
        for(int i=0; i<n; i++)
            System.out.println(array[i]);
    }
}

File: Algorithms/Sorting/BubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;

#define size 10000

int arr[size], n;

void BubbleSort(int arr[], int n) {
	int i, j;
	for (i=0; i<n; ++i) {
		for (j=0; j<n-i-1; ++j) {
			if (arr[j] > arr[j+1]) {
                int tem = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = tem;
			}
		}
	}
}

void getInput() {
    printf("Enter Total number of elements: ");
	scanf("%d", &n);

	printf("Enter %d numbers:\n", n);
	for(int i=0; i<n; i++) {
		scanf("%d", &arr[i]);
	}
}

void showOutput() {
    printf("After Sorting:\n", n);
	for(int i=0; i<n; i++) {
		printf("%d  ", arr[i]);
	} 
}

int main() {
	getInput();
    BubbleSort(arr, n);
    showOutput();

	return 0;
}
